buffer/src/main/java/io/netty/buffer/AdaptivePoolingAllocator.java

```diff
@@ -339,8 +339,8 @@ static int sizeBucket(int size) {
             // so we truncate and roll up the bottom part of the histogram to 8 KiB.
             // The upper size band is 1 MiB, and that gives us exactly 8 size buckets,
             // which is a magical number for JIT optimisations.
-            int normalizedSize = size - 1 >> HISTO_MIN_BUCKET_SHIFT & HISTO_MAX_BUCKET_MASK;
-            return Integer.SIZE - Integer.numberOfLeadingZeros(normalizedSize);
+            int normalizedSize = size - 1 >> HISTO_MIN_BUCKET_SHIFT;
+            return Integer.SIZE - Integer.numberOfLeadingZeros(normalizedSize) & HISTO_MAX_BUCKET_MASK;
```

Comment:
I've noticed that there's no test coverage fo the histogram rotation - do you mind deferring this change in another PR including a test which add such? :pray:

Reply:
@chrisvest Please address this in a followup

Reply:
The allocation test caught this, because the chunks were too small, causing excess allocation. I'll think about how to test this.

Reply:
See https://github.com/netty/netty/pull/14306 and https://github.com/netty/netty/pull/14305